pulumi-deck/src/PulumiStageForm.tsx

```diff
@@ -0,0 +1,214 @@
+import React from 'react';
+
+import {
+  AccountSelectInput,
+  AccountService,
+  FormikFormField,
+  HelpField,
+  IAccountDetails,
+  IFormikStageConfigInjectedProps,
+  noop,
+  ReactSelectInput,
+  SETTINGS,
+  StageConfigField,
+  TextInput,
+} from '@spinnaker/core';
+
+/**
+ * Property names must match the `parameters` defined in the
+ * pre-configured job yaml. The yaml file can be found in the Orca
+ * extension of this plugin.
+ */
+export interface IPulumiProps {
+  gitRepoUrl: string;
+  restoreDependenciesCmd: string;
+  workingDir: string;
+  containerImage: string;
+  pulCommand: string;
+  pulCmdArgs: string;
+  backendUrl: string;
+  k8sSecretsName: string;
+}
+
+/**
+ * Property names must match the `parameters` defined in the
+ * pre-configured job yaml. The yaml file can be found in the Orca
+ * extension of this plugin.
+ */
+export interface IPulumiPluginProps extends IPulumiProps {
+  account: string;
+  namespace: string;
+  stageFieldUpdated?(): void;
+}
+
+export interface IPulumiPluginState {
+  accounts: IAccountDetails[];
+  namespaces: string[];
+}
+
+export class PulumiStageForm extends React.Component<IFormikStageConfigInjectedProps, IPulumiPluginState> {
+  constructor(props: IFormikStageConfigInjectedProps) {
+    super(props);
+
+    this.state = {
+      accounts: [],
+      namespaces: [],
+    };
+  }
+
+  public async componentDidMount(): Promise<void> {
+    if (!this.state.accounts || !this.state.accounts.length) {
+      try {
+        await this.loadAccounts();
+      } catch (err) {
+        console.error('Error loading accounts for kubernetes provider', err);
+      }
+    }
+  }
+
+  private isExpression = (value: string): boolean => (typeof value === 'string' ? value.includes('${') : false);
+
+  public setStateAndUpdateStage(state: Partial<IPulumiPluginState>, cb?: () => void) {
+    this.setState(state as IPulumiPluginState, cb || noop);
```

Comment:
Since there's a default for `cb` what is the case where we need `noop`?

Reply:
Hmm there isn't a default `cb`. Perhaps I am misunderstanding something, but the formal parameter `cb?: () => void` is defining `cb`'s type, no? And not specifying its default value. On the other hand, we could make `noop` a default value of `cb` in the formal param definition itself.

Reply:
Oops -- yes, you are totally right and I misread the parameter. Yes, it's just the typing and wasn't a `= () =>`